test/integration/testsuite/prepare.go

```diff
@@ -0,0 +1,214 @@
+// +build integration
+
+/*
+Copyright (C) 2018 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testsuite
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/minishift/minishift/pkg/minikube/constants"
+	"github.com/minishift/minishift/test/integration/util"
+)
+
+const (
+	delimiterConst = ";"
+)
+
+func ParseFlags() {
+	flag.StringVar(&minishiftArgs, "minishift-args", "", "Arguments to pass to minishift")
+	flag.StringVar(&minishiftBinary, "binary", "", "Path to minishift binary")
+	flag.StringVar(&runBeforeFeature, "run-before-feature", "", "Set of minishift commands to be executed before every feature. Individual commands must be delimited by a semicolon.")
+	flag.StringVar(&testWithShell, "test-with-specified-shell", "", "Name of shell to be used for steps which executes commands directly in persistent shell instance.")
+	flag.StringVar(&copyOcFrom, "copy-oc-from", "", "Path to binary, the binary will $MINISHIFT_HOME/cache/oc.")
+	flag.StringVar(&testDir, "test-dir", "", "Path to the directory in which to execute the tests")
+
+	flag.StringVar(&GodogFormat, "format", "pretty", "Sets which format godog will use")
+	flag.StringVar(&GodogTags, "tags", "", "Tags for godog test")
+	flag.BoolVar(&GodogShowStepDefinitions, "definitions", false, "")
+	flag.BoolVar(&GodogStopOnFailure, "stop-on-failure ", false, "Stop when failure is found")
+	flag.BoolVar(&GodogNoColors, "no-colors", false, "Disable colors in godog output")
+	flag.StringVar(&GodogPaths, "paths", "./features", "")
+
+	flag.Parse()
+}
+
+func HandleISOVersion() {
+	if GodogTags != "" {
+		GodogTags += "&&"
+	}
+	runner := util.MinishiftRunner{CommandPath: minishiftBinary}
+	if runner.IsCDK() {
+		GodogTags += "~minishift-only"
+		isoName = "rhel"
+		fmt.Println("Test run using CDK binary with RHEL iso.")
+	} else {
+		GodogTags += "~cdk-only"
+		isoUrl := os.Getenv("MINISHIFT_ISO_URL")
+		switch isoUrl {
+		case "", "b2d":
+			fmt.Println("Test run using Boot2Docker iso image.")
+			isoName = "b2d"
+		case "minikube":
+			fmt.Println("Test run using Minikube iso image.")
+			isoName = "minikube"
+		case "centos":
+			fmt.Println("Test run using CentOS iso image.")
+			isoName = "centos"
+		default:
+			fmt.Print("Using full path for iso image.")
+			isoName = determineIsoFromFile(isoUrl)
+		}
+	}
+}
+
+func determineIsoFromFile(isoUrl string) string {
```

Comment:
@agajdosi Shouldn't we just validate the IsoURL our and return as it is instead of returning the `b2d, minikube..etc.`. what if I have a custom iso to test and uploaded it some location then integration test might just return `b2d`? I just find it odd.

Reply:
@praveenkumar You can name it to end with `centos7.iso`, then it should be fine.

> Shouldn't we just validate the IsoURL our and return as it is

Later integration tests needs `b2d`, `minikube`, `centos` or `rhel` to determine in some cases during integration tests.

Reply:
too much logic... and I do not see a real need to check if CentOS, Minikube, etc from the prepare code... the test suite should have as little information about this as possible.

Reply:
> the test suite should have as little information about this as possible.

I agree @gbraad, but for example for Docker config I have not found any other way of checking it than cat-ing it. And unfortunately the path differs on all platforms: [catDockerConfigFile()](https://github.com/agajdosi/minishift/blob/dbcf2c4cbb9f396f63d7e5d9c1cd4f8186b5e91a/test/integration/testsuite/testsuite.go#L708).

I suggest to create an issue for the issue you see here and we can then change the code. However for now I am not adding it, instead just moving it.

Reply:
@agajdosi Alright so I understand that some of the integration tests behave as per iso so you need to make sure that shouldn't fail in case of custom ISO URI. Fair enough.

This complete PR is just shuffling code base in a package structure nothing more right?

Reply:
@praveenkumar Yes, it is just a shuffle.

> so you need to make sure that shouldn't fail in case of custom ISO URI

True, currently it is just 1 step which is specific + it is handy for the tag handling, but to be prepared there is this functions so we have the information which can be used if any new test is iso specific.